pkg/unpack/reflector.go

```diff
@@ -0,0 +1,403 @@
+package unpack
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+var zero reflect.Value
+
+type Reflector map[string]map[string]reflect.Type
+
+func New(templates ...interface{}) Reflector {
+	r := Reflector(make(map[string]map[string]reflect.Type))
+	for _, t := range templates {
+		r.Add(t)
+	}
+	return r
+}
+
+func (r Reflector) Add(template interface{}) Reflector {
+	typ := reflect.TypeOf(template)
+	unpackKey, unpackVal, skip, err := structToUnpackRule(typ)
+	if err != nil {
+		panic(err.Error())
+	}
+	if unpackKey == "" {
+		panic(fmt.Sprintf("unpack tag not found for Go type '%s'", typ.Name()))
```

Comment:
`reflect.Type.Name` often returns the empty string for non-struct types. `reflect.Type.String` does not.
```suggestion
		panic(fmt.Sprintf("unpack tag not found for Go type '%s'", typ)))
```